src/content/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/set-max-event-buffer-time.mdx

```diff
@@ -0,0 +1,582 @@
+        <TabsBarItem id="capacitor">
+            Capacitor
+        </TabsBarItem>
+        <TabsBarItem id="cordova">
+            Cordova
+        </TabsBarItem>
+        <TabsBarItem id="flutter">
+            Flutter
+        </TabsBarItem>
+        <TabsBarItem id="react">
+            React Native
+        </TabsBarItem>
+        <TabsBarItem id="xamarin">
+            Xamarin
+        </TabsBarItem>
+        <TabsBarItem id="maui">
+            .NET MAUI
+        </TabsBarItem>
+    </TabsBar>
+
+    <TabsPages>
+        <TabsPageItem id="android">
+## Syntax [#syntax]
+
+### Java [#java]
+
+```java
+NewRelic.setMaxEventBufferTime(int $maxBufferTimeInSec)
+```
+
+### Kotlin [#kotlin]
+
+```kotlin
+NewRelic.setMaxEventBufferTime(maxBufferTimeInSec: Int)
+```
+
+## Description [#description]
+
+Sets the event harvest cycle length.
+
+This method sets the maximum time in seconds that the New Relic Android agent will store events in memory before including the buffered event to be sent to New Relic in the next [harvest cycle](/docs/apm/new-relic-apm/getting-started/glossary#harvest-cycle).
+
+* Default is 600 seconds (10 minutes).
+* Minimum value cannot be less than 60 seconds.
+* Maximum value should not be greater than 600 seconds.
+
+In other words, when the oldest event timestamp exceeds this custom configured time, the agent transmits the buffered content at the harvest cycle between each connection from a New Relic agent to the collector.
+
+See also [`setMaxEventPoolSize()`](/docs/mobile-monitoring/new-relic-mobile-android/android-sdk-api/setmaxeventpoolsize-android-sdk-api), which sets the maximum number of events that are stored by the agent during a harvest cycle.
+
+<Callout variant="important">
+  Be aware that reporting a large number of events, or reporting events too frequently, may impact app performance.
+</Callout>
+
+For context on how to use this API, see [Send custom attributes and events](/docs/mobile-monitoring/new-relic-mobile-android/install-configure/work-android-sdk-api#attributes-events-insights).
+
+## Parameters [#parameters]
+
+<table>
+  <thead>
+    <tr>
+      <th width={200}>
+        Parameter
+      </th>
+      <th width={200}>
+        Type
+      </th>
+      <th>
+        Description
+      </th>
+    </tr>
+  </thead>
+
+  <tbody>
+    <tr>
+      <td>
+        `$maxBufferTimeInSec`
+      </td>
+
+            <td>
+        `int`
+      </td>
+
+      <td>
+        Required. The maximum time (in seconds) that the agent should store events in memory. The default value harvest cycle length is 600 seconds.
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+## Return values [#return-values]
+
+Returns `true` if it succeeds, or `false` if it doesn't.
+
+## Example [#example]
+
+Here's an example to set max event buffer time to 300 seconds:
+
+### Java [#java]
+
+```java
+boolean fiveMinuteLimitSet = NewRelic.setMaxEventBufferTime(300);
+```
+
+### Kotlin [#kotlin]
+
+```kotlin
+
+```
+
```

Comment:
@ndesai-newrelic Can you please provide a kotlin example equivalent to the Java example?

Reply:
NewRelic.setMaxEventBufferTime(300)

Reply:
need to discuss this with you

Reply:
```suggestion
```kotlin

```
   NewRelic.setMaxEventBufferTime(300)
```